### ISSUE TITLE
autopopulate with virtuals

### ISSUE THREAD

**Issue by techaks** [typegoose/typegoose#180]
autopopulate not working with virtual 

mongoose-autopopulate already fixed this bug 

https://github.com/mongodb-js/mongoose-autopopulate/pull/24

**Comment by hasezoey** [comment]
@techaks i dont see a problem, please provide an reproduction script

PS: just to be on the same page, you know that typegoose only has types for `mongoose-autopopulate` and dosnt implement itself?

**Comment by jonasbeyer** [comment]
I've got stuck on the same issue. The plugin doesn't work if the populated field is a virtual.

For example:
```
@prop({
    ref: User,
    localField: "userId",
    foreignField: "_id",
    justOne: true,
    autopopulate: true,
  })
user?: Ref<User>
```

**Comment by hasezoey** [comment]
this should be fixed since 799f0637e725e1d3f766dc682f8601b18f9c8057 (#274)
